modules/infra/datapath/gr_mbuf.h

```diff
@@ -83,3 +83,26 @@ static inline struct rte_mbuf *gr_mbuf_copy(struct rte_mbuf *m, size_t data_len,
 	}
 	return copy;
 }
+
+// Prepend data when possible, and when not, move the entire data forward
+// Slow, but this doesn't require to manage indirect mbufs.
+static inline char *gr_mbuf_prepend(struct rte_mbuf *m, uint16_t len) {
+	char *data = rte_pktmbuf_prepend(m, len);
+	if (unlikely(data == NULL)) {
+		if (rte_pktmbuf_append(m, RTE_PKTMBUF_HEADROOM) != NULL) {
+			data = rte_pktmbuf_mtod(m, void *);
+			memmove(data + RTE_PKTMBUF_HEADROOM,
+				data,
+				rte_pktmbuf_data_len(m) - RTE_PKTMBUF_HEADROOM);
+			rte_pktmbuf_adj(m, RTE_PKTMBUF_HEADROOM);
+			data = rte_pktmbuf_prepend(m, len);
+		}
+	}
+	return data;
+}
```

Comment:
Does that mean that gr_mbuf_prepend() can no longer "fail"?

Maybe we could get rid of all the code where we checked the result of rte_pktmbuf_prepend()?